src/sif/pedestriancost.cc

```diff
@@ -550,7 +550,6 @@ class PedestrianCost : public DynamicCost {
          (edge->use() == baldr::Use::kRailFerry && pred->use() != baldr::Use::kRailFerry);
 
     // Additional penalties without any time cost
-    c.cost += destination_only_penalty_ * (edge->destonly() && !pred->destonly());
```

Comment:
ped with destonly penalty sounds weird:)

Reply:
hm, maybe it has a point since we also turn `private` roads into `destination_only` and even pedestrian shouldn't take those..

Reply:
it's a bit unfortunate though, "residents only" is pointless for ped but we can't distinguish